fhir-model/src/main/java/com/ibm/fhir/model/util/ValidationSupport.java

```diff
@@ -298,6 +300,105 @@ public static void checkLanguageCodeableConcept(CodeableConcept language, String
         }
     }
         
+    /**
+     * Checks that each UCUM code in the list has a valid UCUM syntax.
+     * @param ucumCodes the UCUM code list
+     * @param elementName the element name
+     * @throws IllegalStateException if the passed UCUM code list is not valid
+     */
+    public static void checkUcumCodes(List<Code> ucumCodes, String elementName) {
+        if (ucumCodes != null) {
+            for (Code ucumCode : ucumCodes) {
+                checkUcumCode(ucumCode, elementName);
+            }
+        }
+    }
+    
+    /**
+     * Checks that the UCUM code has a valid UCUM syntax.
+     * @param ucumCode the UCUM code
+     * @param elementName the element name
+     * @throws IllegalStateException if the passed UCUM code is not valid
+     */
+    public static void checkUcumCode(Code ucumCode, String elementName) {
+        if (ucumCode != null) {
+            String ucumCodeValue = ucumCode.getValue();
+            if (!UCUMUtil.isValidUcum(ucumCodeValue)) {
+               throw new IllegalStateException(String.format("'%s' is not a valid UCUM code", ucumCodeValue));
+            }
+        }
+    }
+
+    /**
+     * Checks that each UCUM coding in the list has a valid UCUM syntax.
+     * @param ucumCodings the UCUM coding list
+     * @param elementName the element name
+     * @throws IllegalStateException if the passed UCUM coding list is not valid
+     */
+    public static void checkUcumCodings(List<Coding> ucumCodings, String elementName) {
+        if (ucumCodings != null) {
+            for (Coding ucumCoding : ucumCodings) {
+                checkUcumCoding(ucumCoding, elementName);
+            }
+        }
+    }
+    
+    /**
+     * Checks that the UCUM coding has a valid UCUM syntax.
+     * @param ucumCoding the UCUM coding
+     * @param elementName the element name
+     * @throws IllegalStateException if the passed UCUM coding is not valid
+     */
+    public static void checkUcumCoding(Coding ucumCoding, String elementName) {
+        if (ucumCoding != null) {
+            if (hasSystemAndCodeValues(ucumCoding)) {
+                if (!UCUM_CODE_SYSTEM_URL.equals(ucumCoding.getSystem().getValue())) {
```

Comment:
This could be refactored a little into one if statement and the else could be the checkUcumCode (it feels super nested)  also ucumCoding.getSystem is this ever null on 355? maybe just combine this with 354 and 355

Reply:
I need the if check on line 353 by itself b/c I don't want to return an exception if the coding is null. The if check on line 354 is just to verify that I have values for both system and code. If I don't, I can skip validating those fields separately since I know the coding is not valid without both system and code values. I can move the checkUcumCode to an else leg for the if check on line 355 (ucumCoding.getSystem will never be null here due to the checking in line above in the hasSystemAndCodeValues method)